app/src/main/java/org/wikipedia/readinglist/db/ReadingListDao.kt

```diff
@@ -52,7 +52,7 @@ interface ReadingListDao {
         val lists = getListsWithoutContents()
         val pages = AppDatabase.instance.readingListPageDao().getAllPagesToBeSynced()
         pages.forEach { page ->
-            lists.first { it.id == page.listId }.apply { this.pages.add(page) }
```

Comment:
If the database contains an orphaned page (not belonging to any reading list, possibly caused by a crash while syncing), then this will crash silently, and prevent syncing from continuing.